src/librustc_mir/interpret/eval_context.rs

```diff
@@ -1295,6 +1296,96 @@ impl<'a, 'tcx, M: Machine<'tcx>> EvalContext<'a, 'tcx, M> {
         }
     }
 
+    pub fn read_discriminant_value(
```

Comment:
Why were these functions moved?

Reply:
Why should they be in terminator/mod.rs? That doesn't make any sense.

eval_context has way too much stuff though, so maybe there's a better place to put them?

Reply:
Sounds like a good start. Should however probably be a separate PR.